apps/web/src/components/Feed/FeedItem.tsx

```diff
@@ -0,0 +1,174 @@
+import { useCallback } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import breakpoints from '~/components/core/breakpoints';
+import { VStack } from '~/components/core/Spacer/Stack';
+import { FEED_EVENT_ROW_WIDTH_DESKTOP } from '~/components/Feed/dimensions';
+import { FeedEventData } from '~/components/Feed/FeedEventData';
+import { FeedEventSocializeSection } from '~/components/Feed/Socialize/FeedEventSocializeSection';
+import { FeedMode } from '~/components/Feed/types';
+import { FeedItemFragment$key } from '~/generated/FeedItemFragment.graphql';
+import { FeedItemQueryFragment$key } from '~/generated/FeedItemQueryFragment.graphql';
+import { FeedItemWithErrorBoundaryFragment$key } from '~/generated/FeedItemWithErrorBoundaryFragment.graphql';
+import { FeedItemWithErrorBoundaryQueryFragment$key } from '~/generated/FeedItemWithErrorBoundaryQueryFragment.graphql';
+import { ReportingErrorBoundary } from '~/shared/errors/ReportingErrorBoundary';
+import { TriedToRenderUnsupportedFeedEvent } from '~/shared/errors/TriedToRenderUnsupportedFeedEvent';
+
+import PostData from './Posts/PostData';
+import PostSocializeSection from './Socialize/PostSocializeSection';
+
+type FeedItemProps = {
+  eventRef: FeedItemFragment$key;
+  queryRef: FeedItemQueryFragment$key;
+  feedMode: FeedMode;
+};
+
+// Not to be confused with the FeedEvent type. This component can render both FeedEvent and Post types, and represents a single item in the feed.
+function FeedItem({ eventRef, queryRef, feedMode }: FeedItemProps) {
```

Comment:
renamed this component from FeedEvent to FeedItem and refactored so that it supports both FeedEvent and Post types